### PR TITLE
Signing component

### PR DESCRIPTION
**Pull request by TommyHuang-dev**

### CODE REVIEW

File: src/app/pages/signing/signing.component.css [hunk 1]
```diff
@@ -1,17 +1,68 @@
...
+
+
+#signing-box {
+	margin: 1em;
+	padding: 1em;
+	width: 50em;
+	resize: none;
+	font: 16px Arial;
+	border-width: 0.05em;
+	border-radius: 0em;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.css:38]: Unnecessary units
```suggestion
	border-radius: 0;
```

File: src/app/pages/signing/signing.component.css [hunk 2]
```diff
@@ -1,17 +1,68 @@
...
+}
+
+.form-control:focus {
+	border-color: var(--accent);
+	box-shadow: none;
+}
+
+#preview-header {
+	margin-top: 1.0em;
+	margin-bottom: 0em;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.css:48]: Unnecessary units
```suggestion
	margin-bottom: 0;
```

File: src/app/pages/signing/signing.component.css [hunk 3]
```diff
@@ -1,17 +1,68 @@
...
+	padding: 1em;
+	width: 50em;
+	height: 20em;
+	resize: none;
+	font: 16px Arial;
+	border: 0.05em solid var(--light-grey);
+	overflow-y: auto;
+	overflow-x: auto;
+  word-wrap: break-word;
+  border-radius: 0em;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.css:63]: Unnecessary units
```suggestion
  border-radius: 0;
```

File: src/app/pages/signing/signing.component.css [hunk 4]
```diff
@@ -1,17 +1,68 @@
...
+
+#fontSelectorDropdown {
+	text-align: right;
+	width: 12em;
 }
 
 .format-icons {
-  width: 1.8em;
-  height: 1.8em;
+	width:1.8em;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.css:15]: Missing space
```suggestion
	width: 1.8em;
```

File: src/app/pages/signing/signing.component.css [hunk 5]
```diff
@@ -1,17 +1,68 @@
...
+	height:1.8em;
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.css:16]: Missing space
```suggestion
	height: 1.8em;
```

File: src/app/pages/signing/signing.component.css (hunk 1 above)
Comment (GuyInFridge) [inline src/app/pages/signing/signing.component.css:38]:
```suggestion
	border-radius: 0;
```
Remove this elsewhere tooo

File: src/app/pages/signing/signing.component.html [hunk 6]
```diff
@@ -1,61 +1,73 @@
...
-
-    <div class="btn-group mr-2" role="group" aria-label="Font group">
-      <button type="button" class="btn btn-primary">Font</button>
-      <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-toggle="dropdown"
-              aria-haspopup="true" aria-expanded="false">
-        <span class="sr-only">Toggle Dropdown</span>
+  <div class="row justify-content-center" id="options">
+    <div class="dropdown">
+      <button class="btn btn-secondary dropdown-toggle" type="button" id="fontSelectorDropdown" data-toggle="dropdown"
+        aria-haspopup="true" aria-expanded="false">
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:20]: Not enough indent
```suggestion
              aria-haspopup="true" aria-expanded="false">
```

File: src/app/pages/signing/signing.component.css [hunk 7]
```diff
@@ -1,17 +1,68 @@
...
+	font-size: 1.2em;
+}
+
+markdown {
+	margin: 1em;
+	padding: 1em;
+	width: 50em;
+	height: 20em;
+	resize: none;
+	font: 16px Arial;
```
Comment (GuyInFridge) [inline src/app/pages/signing/signing.component.css:58]: All sizes should be `em`, not `px`.

File: src/app/pages/signing/signing.component.html [hunk 8]
```diff
@@ -1,61 +1,73 @@
...
+  <div class="row" id="title-div">
+    <label for="signing-box">
+      <h2 id="signing-header">Type your message to [friend name] below.</h2>
+    </label>
   </div>
-  <div class="row">
-    <textarea rows="8" class="form-control" id="signing-box" autofocus [(ngModel)]='signingBoxText'></textarea>
+  <div class="d-flex justify-content-center col-12">
+    <div class="row">
+      <textarea #inputarea rows="6" class="form-control" id="signing-box" autofocus [(ngModel)]='signingBoxText'
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:9]: You should rename `inputarea` to `inputArea`

File: src/app/pages/signing/signing.component.html [hunk 9]
```diff
@@ -1,61 +1,73 @@
...
-    <div class="btn-toolbar" role="toolbar" aria-label="Toolbar for formatting text"></div>
-
-    <div class="btn-group mr-2" role="group" aria-label="Font group">
-      <button type="button" class="btn btn-primary">Font</button>
-      <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-toggle="dropdown"
-              aria-haspopup="true" aria-expanded="false">
-        <span class="sr-only">Toggle Dropdown</span>
+  <div class="row justify-content-center" id="options">
+    <div class="dropdown">
+      <button class="btn btn-secondary dropdown-toggle" type="button" id="fontSelectorDropdown" data-toggle="dropdown"
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:19]: Improper ID name
```suggestion
      <button class="btn btn-secondary dropdown-toggle" type="button" id="font-selector-dropdown" data-toggle="dropdown"
```

File: src/app/pages/signing/signing.component.html [hunk 10]
```diff
@@ -1,61 +1,73 @@
...
+        aria-haspopup="true" aria-expanded="false">
+        {{ currentFont }}
       </button>
-      <div class="dropdown-menu">
-        <a class="dropdown-item" href="javascript: void(0)">Arial</a>
-        <a class="dropdown-item" href="javascript: void(0)">Comic Sans</a>
-        <a class="dropdown-item" href="javascript: void(0)">Times New Roman</a>
+      <div class="dropdown-menu" aria-labelledby="fontSelectorDropdown">
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:23]: Improper ID name
```suggestion
      <div class="dropdown-menu" aria-labelledby="font-selector-dropdown">
```

File: src/app/pages/signing/signing.component.html [hunk 11]
```diff
@@ -1,61 +1,73 @@
-<div class="container py-5 center-children vertical-center">
-  <div class="row">
-    <h2>Sending Message to friend.</h2>
+<div class="container py-5 d-flex vertical-center">
+  <div class="row" id="title-div">
+    <label for="signing-box">
+      <h2 id="signing-header">Type your message to [friend name] below.</h2>
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:4]: The `<h2>` tag is not allowed in `<label>`. You should create a CSS class on the `<label>` to format the text.

File: src/app/pages/signing/signing.component.html [hunk 12]
```diff
@@ -1,61 +1,73 @@
...
-    autofocus
-  ></textarea>
-  <div class="row">
-    <markdown id="preview-box" data="{{signingBoxText}}"></markdown>
-  </div>
 </div>
 
+
+
+
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:68]: Too many blank lines
```suggestion
```

File: src/app/pages/signing/signing.component.html [hunk 13]
```diff
@@ -1,61 +1,73 @@
...
-    id="signing-box"
-    autofocus
-  ></textarea>
-  <div class="row">
-    <markdown id="preview-box" data="{{signingBoxText}}"></markdown>
-  </div>
 </div>
 
+
+
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:67]: Too many blank lines
```suggestion
```

File: src/app/pages/signing/signing.component.html [hunk 14]
```diff
@@ -1,61 +1,73 @@
...
+
 <div class="modal fade" id="submitConfirmModal" tabindex="-1" role="dialog" aria-labelledby="submitConfirmModalTitle"
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:69]: Improper ID name
**Also change this in Line 74**
```suggestion
<div class="modal fade" id="submit-confirm-modal" tabindex="-1" role="dialog" aria-labelledby="submit-confirm-modal-title"
```

File: src/app/pages/signing/signing.component.html [hunk 15]
```diff
@@ -1,61 +1,73 @@
...
       </div>
       <div>
-        <input type="color" id="colorpicker mr-2" name="fontColorPicker" value="#000000">
+        <input type="color" class="mr-2" id="colorpicker-background" name="backgroundColorPicker"
+          value={{backgroundColor}} [(ngModel)]='backgroundColor'>
       </div>
     </div>
-    <button type="button" class="btn btn-secondary" data-toggle="modal" data-target="#submitConfirmModal">
-      Submit
+    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#submitConfirmModal">
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:60]: Improper ID Name
```suggestion
    <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#submit-confirm-modal">
```

File: src/app/pages/signing/signing.component.html [hunk 16]
```diff
@@ -73,4 +85,4 @@ <h5 class="modal-title" id="submitConfirmModalTitle">Are you sure you want to su
       </div>
     </div>
   </div>
-</div>
+</div>
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:88]: No newline at end of file
```suggestion
</div>

```

File: src/app/pages/signing/signing.component.ts [hunk 17]
```diff
@@ -8,9 +8,44 @@ import { Component, OnInit } from '@angular/core';
...
 export class SigningComponent implements OnInit {
   public signingBoxText = '';
+  // fonts and colors
+  public currentFont = 'Arial';
+  public backgroundColor = '#ffffff';
+  public textColor = '#000000';
+
+  selectFont(font: string) {
+    this.currentFont = font;
+  }
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:18]: Add blank line after function
```suggestion
  }
  
```

File: src/app/pages/signing/signing.component.ts [hunk 18]
```diff
@@ -8,9 +8,44 @@ import { Component, OnInit } from '@angular/core';
...
+  makeBold(textbox: HTMLInputElement) {
+    const start = textbox.selectionStart;
+    const end = textbox.selectionEnd;
+    const text = textbox.value;
+    this.signingBoxText = text.slice(0, start) + '**' + text.slice(start, end) + '**' + text.slice(end);
+    textbox.select();
+  }
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:25]: Add blank line after function
```suggestion
  }
  
```

File: src/app/pages/signing/signing.component.ts [hunk 19]
```diff
@@ -8,9 +8,44 @@ import { Component, OnInit } from '@angular/core';
...
+  makeItalics(textbox: HTMLInputElement) {
+    const start = textbox.selectionStart;
+    const end = textbox.selectionEnd;
+    const text = textbox.value;
+    this.signingBoxText = text.slice(0, start) + '*' + text.slice(start, end) + '*' + text.slice(end);
+    textbox.select();
+  }
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:32]: Add blank line after function
```suggestion
  }
  
```

File: src/app/pages/signing/signing.component.ts [hunk 20]
```diff
@@ -8,9 +8,44 @@ import { Component, OnInit } from '@angular/core';
...
+  makeUnderline(textbox: HTMLInputElement) {
+    const start = textbox.selectionStart;
+    const end = textbox.selectionEnd;
+    const text = textbox.value;
+    this.signingBoxText = text.slice(0, start) + '<u>' + text.slice(start, end) + '</u>' + text.slice(end);
+    textbox.select();
+  }
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:39]: Add blank line after function
```suggestion
  }
  
```

File: src/app/pages/signing/signing.component.html (hunk 8 above)
Comment (janakitti) [reply src/app/pages/signing/signing.component.html:9]: There are also other instances of 'inputarea' in this file just so you know.

**Pull request merged**